database/src/lib.rs

```diff
@@ -1009,4 +1013,73 @@ impl BenchmarkRequestIndex {
     pub fn completed_requests(&self) -> &HashSet<String> {
         &self.completed
     }
+
+    pub fn add_tag(&mut self, tag: &str) {
+        self.completed.insert(tag.to_string());
```

Comment:
This should also add the tag to `all`, if we're trying to keep the cache consistent.